File: js/0_hello/enum.js
// JAVASCRIPTの擬似列挙型
var FRUITS_TYPE = {
    UNKNOWN : 0,
    BANANA : 1,
    APPLE : 2,
    ORANGE : 3,
    WATERMELON : 4,
};
FRUITS_TYPE.SIZEOF = 5;

function init(l){
    for(var i=0;i<FRUITS_TYPE.SIZEOF;++i){
        switch(i){
        case FRUITS_TYPE.BANANA:
            l[i] = 'BAD';
            break;
        case FRUITS_TYPE.APPLE:
            l[i] = 'GOOD';
            break;
        case FRUITS_TYPE.ORANGE:
            l[i] = 'GOOD';
            break;
        case FRUITS_TYPE.WATERMELON:
            l[i] = 'GOOD';
            break;
        default :
            l[i] = 'UNKOWN';
            break;
        }
    }
}
var list = new Array(FRUITS_TYPE.SIZEOF);
init(list);
console.log('FAVORITE BANANA?');
console.log('BANANA is '+list[FRUITS_TYPE.BANANA]);

File: c/0_hello/enum.c
#include <stdio.h>

typedef enum{
    FRUITS_TYPE_UNKNOWN = 0,
    FRUITS_TYPE_BANANA,
    FRUITS_TYPE_APPLE,
    FRUITS_TYPE_ORANGE,
    FRUITS_TYPE_WATERMELON,

    SIZEOF_FRUITS_TYPE,
}FRUITS_TYPE;

void init(char **list){
    int i;
    for(i=0;i<SIZEOF_FRUITS_TYPE;++i){
        switch(i){
        case FRUITS_TYPE_BANANA:
            list[i] = "BAD";
            break;
        case FRUITS_TYPE_APPLE:
            list[i] = "GOOD";
            break;
        case FRUITS_TYPE_ORANGE:
            list[i] = "GOOD";
            break;
        case FRUITS_TYPE_WATERMELON:
            list[i] = "GOOD";
            break;
        default :
            list[i] = "UNKOWN";
            break;
        }
    }
}

int main(int argc, char *argv[]){

    char *list[ SIZEOF_FRUITS_TYPE ];
    init(list);

    printf("%s\n","FAVORITE BANANA?");
    printf("BANANA is %s\n", list[FRUITS_TYPE_BANANA]);
    return 0;
}

